pennylane/ftqc/qubit_graph.py

```diff
@@ -0,0 +1,401 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module contains the data structures to represent the qubit memory model in the FTQC/MBQC
+framework, as well as the API to access and manipulate these data structures.
+"""
```

Comment:
```suggestion
"""This module contains the data structures to represent the qubits using a hierarchical memory model, as well as defining the API to access and manipulate these data structures.
"""
```

I think we can likely define this without the MBQC/FTQC points. We can likely use these structures independent of FTQC too (e.g. replacemenbt blocks for applying algorithms, etc/)